src/overlays/actors/ovl_En_Horse_Normal/z_en_horse_normal.c

```diff
@@ -21,45 +42,635 @@ const ActorInit En_Horse_Normal_InitVars = {
+                }
+                break;
+            case 1:
+            case 2:
+            case 3:
+                break;
+        }
+
+        this->unk_1E4 &= ~1;
+        this->unk_1E4 &= ~2;
+        if (phi_t0 == 1) {
+            Audio_PlaySoundGeneral(NA_SE_EV_HORSE_GROAN, &this->unk_204, 4, &D_801333E0, &D_801333E0, &D_801333E8);
+        } else if (phi_t0 == 3) {
+            Audio_PlaySoundGeneral(NA_SE_EV_HORSE_NEIGH, &this->unk_204, 4, &D_801333E0, &D_801333E0, &D_801333E8);
+        } else {
+            func_80A6BCEC(this);
+        }
+        if (phi_t0 != this->animationIdx) {
+            this->animationIdx = phi_t0;
+            SkelAnime_ChangeAnim(&this->skin.skelAnime, sAnimations[this->animationIdx], func_80A6B30C(this), 0.0f,
+                                 SkelAnime_GetFrameCount(&sAnimations[this->animationIdx]->genericHeader), 2, -3.0f);
+        } else {
+            SkelAnime_ChangeAnim(&this->skin.skelAnime, sAnimations[this->animationIdx], func_80A6B30C(this), 0.0f,
+                                 SkelAnime_GetFrameCount(&sAnimations[this->animationIdx]->genericHeader), 2, 0.0f);
+        }
+    }
+}
+
+void func_80A6C4CC(EnHorseNormal* this) {
+    this->type = 2;
+    this->animationIdx = 0;
+    this->unk_21C = 0;
+    this->unk_21E = 0;
+    this->actor.speedXZ = 0.0f;
+    this->unk_218 = 0.0f;
+    SkelAnime_ChangeAnim(&this->skin.skelAnime, sAnimations[this->animationIdx], func_80A6B30C(this), 0.0f,
+                         SkelAnime_GetFrameCount(&sAnimations[this->animationIdx]->genericHeader), 2, 0.0f);
+}
+
+void EnHorseNormal_Wait(EnHorseNormal* this, GlobalContext* globalCtx) {
+    if (SkelAnime_FrameUpdateMatrix(&this->skin.skelAnime)) {
+        f32 rand = Math_Rand_ZeroOne();
+
+        if (rand < 0.4f) {
+            this->animationIdx = 0;
+        } else if (rand < 0.8f) {
+            this->animationIdx = 1;
+            Audio_PlaySoundGeneral(NA_SE_EV_HORSE_GROAN, &this->unk_204, 4, &D_801333E0, &D_801333E0, &D_801333E8);
+        } else {
+            this->animationIdx = 3;
+            Audio_PlaySoundGeneral(NA_SE_EV_HORSE_NEIGH, &this->unk_204, 4, &D_801333E0, &D_801333E0, &D_801333E8);
+        }
+
+        SkelAnime_ChangeAnim(&this->skin.skelAnime, sAnimations[this->animationIdx], func_80A6B30C(this), 0.0f,
+                             SkelAnime_GetFrameCount(&sAnimations[this->animationIdx]->genericHeader), 2, 0.0f);
+    }
+}
+
+void func_80A6C6B0(EnHorseNormal* this) {
+    this->type = 3;
+    this->animationIdx = 0;
+    this->unk_21C = 0;
+    this->unk_21E = 0;
+    this->actor.flags |= 0x30;
+    this->actor.speedXZ = 0.0f;
+    this->unk_218 = 0.0f;
+    SkelAnime_ChangeAnim(&this->skin.skelAnime, sAnimations[this->animationIdx], func_80A6B30C(this), 0.0f,
+                         SkelAnime_GetFrameCount(&sAnimations[this->animationIdx]->genericHeader), 2, 0.0f);
+}
+
+void EnHorseNormal_WaitClone(EnHorseNormal* this, GlobalContext* globalCtx) {
```

Comment:
naming action functions is hard but from what I saw "Wait" is used when the action will eventually change due to a timer or events (switch flag, colliding)
The name also implies "waiting for clone" imo instead of "I am cloned and idling"? "ClonedIdle"?

But does this function actually do something cloning-related? idk if "Clone" should be included in the name only because the only type using this function is the cloned type
can't really tell the difference from `EnHorseNormal_Wait` though because of the unks

Reply:
I think we may need some input from others on this one. I do agree that `Idle` would make more sense (to me at least) though @fig02 said we don't usually use the word `Idle` for naming action functions. Also, I think we usually put a verb first in the name so `IdleCloned` might be better. Everyone please let me know your thoughts.